src/schema/v2/infiniteDiscovery/discoverArtworks.ts

```diff
@@ -47,21 +49,76 @@ export const DiscoverArtworks: GraphQLFieldConfig<void, ResolverContext> = {
         },
       }),
     },
+    useRelatedArtworks: { type: GraphQLBoolean, defaultValue: false },
   }),
   resolve: async (
     _root,
     args,
-    { weaviateCreateObjectLoader, weaviateGraphqlLoader, artworksLoader }
+    {
+      weaviateCreateObjectLoader,
+      weaviateGraphqlLoader,
+      artworksLoader,
+      relatedArtworksLoader,
+      marketingCollectionLoader,
+      savedArtworksLoader,
+    }
   ) => {
     if (
+      !artworksLoader ||
       !weaviateGraphqlLoader ||
-      !weaviateCreateObjectLoader ||
-      !artworksLoader
+      !weaviateCreateObjectLoader
     ) {
       new Error("A loader is not available")
     }
 
-    const { userId, limit = 5, offset = 0, certainty = 0.5, sort } = args
+    const {
+      userId,
+      limit = 5,
+      offset = 0,
+      certainty = 0.5,
+      sort,
+      useRelatedArtworks,
+    } = args
+
+    if (useRelatedArtworks) {
+      if (!savedArtworksLoader) {
+        return new Error("Saved artworks loader is not available")
+      }
+
+      const { body: savedArtworks } = await savedArtworksLoader({
+        size: 28,
+        sort: "-position",
+        user_id: userId,
```

Comment:
I want to call out that we've been pretty loose up to this point about "userId", as its made it simpler to test weaviate just using userid's of `1`, or `2`, `3`,.... or even an `email`. If we want to share the same loader for testing we may need consider this.

Reply:
That’s a good point. For this phase, we could support both parameters. My understanding is that we still want to explore various options and consider this and other implementation details later when finalizing the project. Ultimately, I believe we want to avoid using `userId` and move this functionality to the Me type. What's your take on that?

Reply:
>For this phase, we could support both parameters. My understanding is that we still want to explore various options and consider this and other implementation details later when finalizing the project.

That works, but want to make sure we aren't so loose we make our own lives more difficult. I think I have probably been too quick to just shove whatever I need into the `userId` field. I would like to take a refactor pass before the end of my rotation at the end of the week. 

>Ultimately, I believe we want to avoid using userId and move this functionality to the Me type.

Yea, that makes sense to me.

Reply:
> That works, but want to make sure we aren't so loose we make our own lives more difficult. I think I have probably been too quick to just shove whatever I need into the userId field. I would like to take a refactor pass before the end of my rotation at the end of the week.

Sounds good, let's do that after the broader testing? Not sure if breaking changes to the schema would be something we want to do before that? 😅